src/test/RwaUrn.t.sol

```diff
@@ -329,6 +329,107 @@ contract RwaExampleTest is DSTest, DSMath, TryPusher {
         assertEq(dai.balanceOf(address(inConduit)), 0 ether);
     }
 
+    function test_partial_repay_fuzz(uint256 drawAmount, uint256 wipeAmount, uint256 drawTime, uint256 wipeTime) public {
+        // Convert to reasonable numbers
+        drawAmount = (drawAmount % 300 ether) + 100 ether;      // 100-400 ether
+        wipeAmount = wipeAmount % drawAmount;                   // 0-drawAmount ether
+        drawTime = drawTime % 15 days;                       // 0-15 days
+        wipeTime = wipeTime % 15 days;                       // 0-15 days
+
+        usr.lock(1 ether);
+
+        hevm.warp(now + drawTime);
+        jug.drip("acme");
+
+        usr.draw(drawAmount);
+
+        // usr nominates ultimate recipient
+        usr.pick(address(rec));
+        outConduit.push();
+
+        hevm.warp(now + wipeTime);
+        jug.drip("acme");
+
+        rec.transfer(address(inConduit), wipeAmount);
+        assertEq(dai.balanceOf(address(inConduit)), wipeAmount);
+
+        inConduit.push();
+        usr.wipe(wipeAmount);
+    }
+
+    function test_repay_rounding_fuzz(uint256 drawAmt, uint256 drawTime, uint256 wipeTime) public {
+        // Convert to reasonable numbers
+        drawAmt = (drawAmt % 300 ether) + 99.99 ether; // 99.99-399.99 ether
+        drawTime = drawTime % 15 days;                       // 0-15 days
+        wipeTime = wipeTime % 15 days;                       // 0-15 days
+
+        (uint256 ink, uint256 art) = vat.urns("acme", address(urn));
+        assertEq(ink, 0);
+        assertEq(art, 0);
+
+        usr.lock(1 ether);
+
+        hevm.warp(now + drawTime);
+        jug.drip("acme");
+
+        usr.draw(drawAmt);
+
+        uint256 urnVatDust = vat.dai(address(urn));
+
+        // A draw should leave less than 2 RAY dust
+        assertTrue(urnVatDust < 2 * RAY);
+
+        (, uint256 rate,,,) = vat.ilks("acme");
+        (ink, art) = vat.urns("acme", address(urn));
+        assertEq(ink, 1 ether);
+        assertEq(art, (rad(drawAmt) + urnVatDust) / rate);
+
+        // usr nominates ultimate recipient
+        usr.pick(address(rec));
+        outConduit.push();
+
+        hevm.warp(now + wipeTime);
+        jug.drip("acme");
+
+        (, rate,,,) = vat.ilks("acme");
+
+        uint256 fullWipeAmt = art * rate / RAY;
+        if (fullWipeAmt * RAY < art * rate) {
+            fullWipeAmt += 1;
+        }
+
+        // Forcing extra DAI balance to pay accumulated fee
+        hevm.store(
+            address(dai),
+            keccak256(abi.encode(address(rec), uint256(3))),
```

Comment:
Need to figure it out why in this case the storage slot seems to have an offset space. For some reason the balance seems to be in number 3 and the total supply in 2, when it should be 2 and 1.

Reply:
Nvm, just realized it is a `DSToken` what being used in the unit test. Maybe we want to use the proper Dai contract.